### ISSUE TITLE
feature: spring-boot2 support for retry

### ISSUE THREAD

**Issue by davidkarlsen** [resilience4j/resilience4j#348]
It would be nice to have spring-boot2-iness for the retry functionality:
* autoconfig
* configure retry-settings through yaml
* apply in code as annotation + aspect like for @CircuitBreaker

**Comment by RobWin** [comment]
Would you like to create a PR?

**Comment by Romeh** [comment]
@RobWin  I am working on adding Sync retry spring-boot2-iness  support , once it is done locally i will create PR for it also a possible question if we want to support the Async retry ?

**Comment by Romeh** [comment]
@RobWin  @davidkarlsen  https://github.com/resilience4j/resilience4j/pull/358 a pull request has been created , can u please check it out and would be appreciated to do some reviews there as well :)

**Comment by davidkarlsen** [comment]
Looks like this was merged. Could you do a release too?

**Comment by davidkarlsen** [comment]
@RobWin should this be closed now? Is there anything else missing?

**Comment by RobWin** [comment]
@davidkarlsen Did you have time to check if the SNAPSHOT version is working for you or is anything missing?

**Comment by davidkarlsen** [comment]
I never tested it but can give it a spin. Do you push out snapshots anywhere?

**Comment by RobWin** [comment]
Yes, http://resilience4j.github.io/resilience4j/#_gradle

**Comment by dannycohn** [comment]
I'd like to try out the snapshot to evaluate retry annotations. I don't see a recent snapshot in the snapshot repo.  Am I missing something?

**Comment by RobWin** [comment]
No you are right. I've to check it.

**Comment by dannycohn** [comment]
Thanks @RobWin. Looking forward to trying it out. We're getting ready to implement resilience4j. Do you have any guess on when you might issue a release? I'm wondering if I'lll have to go fight to get approval to use a SNAPSHOT or if  you'll have a release before we start integration on our end.

**Comment by RobWin** [comment]
It's working again.
Please see: https://oss.jfrog.org/artifactory/oss-snapshot-local/io/github/resilience4j/resilience4j-spring-boot2/0.14.0-SNAPSHOT/

**Comment by Fiazz** [comment]
Thanks @Robwin. Will try it out.

**Comment by RobWin** [comment]
I can release 0.14.0 as soon as it is working for you as expected.

**Comment by Fiazz** [comment]
Yes @RobWin ,Retry worked well for my expectations.

**Comment by dannycohn** [comment]
Alas, I have not been able to try this myself yet.  I tried, but getting the snapshots working with my setup was proving complex.  I can try again in a couple days, or 🤞 that @Fiazz's feedback is enough to get 0.14.0 released 😄

**Comment by davidkarlsen** [comment]
where are the updated docs? I'm giving this a spin now

**Comment by RobWin** [comment]
We have to refactor it a little bit. Work is still ongoing so that the Retry AOP Aspect can also handle CompletionStage, RxJava and Reactor return types.

**Comment by RobWin** [comment]
0.14.0 was released
